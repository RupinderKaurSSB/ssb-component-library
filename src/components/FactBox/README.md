FactBox
========

> Wrapper for accordion

### Usage

#### HTML

```html
<div class="ssb-fact-box">
	<div class="ssb-accordion">
	  <div class="accordion-header closed" onclick="{toggle classname to 'open'}">
		{plus icon}
		<div class="ssb-title header-text no-margin"><h5>Title</h5></div>
	  </div>
	  <div class="accordion-body closed">
		{insert content}
	  </div>
	</div>
</div>
```

#### React

```jsx harmony
<FactBox header="This is a header" text="This is paragraph text which explains the accordion" />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| header | string | Header text |
| text | string or element | Content in fact box |
| openByDefault | bool | Open when rendered. Defaults to false |
