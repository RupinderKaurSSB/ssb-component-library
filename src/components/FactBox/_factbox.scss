.ssb-fact-box {
  border: 1px solid $ssb-green-3;
  margin: 16px 0;
  padding: 20px 16px;
  width: 100%;

  &:hover {
    border: 1px solid $ssb-green-4;
  }
}
