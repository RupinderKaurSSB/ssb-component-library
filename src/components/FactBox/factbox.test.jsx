import React from 'react';
import {shallow} from 'enzyme';
import FactBox from './index';

describe('FactBox component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<FactBox header="fact box header" text="fact box Text"  />);
		expect(wrapper).toMatchSnapshot();
	});
	test('set openByDefault', () => {
		const wrapper = shallow(<FactBox header="fact box header" text="fact box Text" openByDefault />);
		expect(wrapper.find('Accordion').props().openByDefault).toEqual(true);
	});

});
