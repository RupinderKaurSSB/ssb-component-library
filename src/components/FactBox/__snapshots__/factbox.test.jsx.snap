// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactBox component Matches the snapshot 1`] = `
<div
  className="ssb-fact-box"
>
  <Accordion
    header="fact box header"
    openByDefault={false}
    primary={true}
    tabIndex={0}
  >
    fact box Text
  </Accordion>
</div>
`;
